src/lib.rs

```diff
@@ -56,6 +61,74 @@ impl Reading {
     }
 }
 
+#[derive(Debug)]
+struct Output {
+    description: String,
+    error: Option<reqwest::Error>,
+    data: Option<Response>,
+    thread_id: ThreadId,
+    debug: bool
+  }
+
+impl Output {
+    pub fn new(
+        description: String,
+        error: Option<reqwest::Error>,
+        data: Option<Response>,
+        thread_id: ThreadId,
+        debug: bool
+    ) -> Output {
+        Output {
+            description,
+            error,
+            data,
+            thread_id,
+            debug
+        }
+    }
+
+    pub fn is_error(&self) -> bool {
+      self.error.is_some()
```

Comment:
I absolutely love the use of Rust's `Option` concept here, fits perfectly in line with success vs error cases.